pandas/tests/series/test_dtypes.py

```diff
@@ -153,6 +153,38 @@ def test_astype_dict(self):
         self.assertRaises(KeyError, s.astype, {'abc': str, 'def': str})
         self.assertRaises(KeyError, s.astype, {0: str})
 
+    def test_astype_generic_timestamp(self):
+        # see gh-15524
+        data = [1]
+
+        s = Series(data)
+        dtype = np.datetime64
+        result = s.astype(dtype)
+        expected = Series(data, dtype=dtype)
+        assert_series_equal(result, expected)
+
+        s = Series(data)
+        dtype = np.timedelta64
+        result = s.astype(dtype)
+        expected = Series(data, dtype=dtype)
+        assert_series_equal(result, expected)
+
+    def test_astype_empty_constructor_equality(self):
+        # see gh-15524
+
+        for dtype in np.typecodes['All']:
```

Comment:
But wait...you're okay with leaving a mixture of unittest and pytest idioms?

What you've described is exactly why I don't want to bend over backwords and use the pytest idiom for this one paramterization.  Nosetests used to be that framework, and now we will have to rewrite all of our tests to align with the pytest idiom.

The testing interface should be test-framework agnostic is what I'm saying so that all we have to do is change the implementation of that interface and not have to deal with this idiom burden that we have now.